### PR TITLE
Fixes/bugs

### LINKED ISSUE #48: Unable to open database file in read-only mode. Ensure that the database exists in order to use read-only mode.
After latest docker update got the message: Unable to open database file in read-only mode. Ensure that the database exists in order to use read-only mode.

Debug log is here:

```
Fri, 17 Dec 2021 08:46:20 INFO Log level: DEBUG
Fri, 17 Dec 2021 08:46:20 INFO Setting timzeone to Europe/Kiev
Fri, 17 Dec 2021 08:46:20 INFO Setting gid to '1000'.
Fri, 17 Dec 2021 08:46:20 INFO Setting uid to '100'.
Fri, 17 Dec 2021 08:46:20 INFO Storing database file to /home/py-kms/db/pykms_database.db
Fri, 17 Dec 2021 08:46:20 DEBUG server_cmd: /usr/bin/python3 -u pykms_Server.py 0.0.0.0 1688 -l 1033 -c 26 -a 120 -r 10080 -w RANDOM -V DEBUG -F STDOUT -s /home/py-kms/db/pykms_database.db
Fri, 17 Dec 2021 08:46:21 INFO TCP server listening at 0.0.0.0 on port 1688
Fri, 17 Dec 2021 08:46:21 INFO HWID: E6C388F490944342
Fri, 17 Dec 2021 08:46:25 INFO Starting a dummy activation to ensure the database file is created
Fri, 17 Dec 2021 08:46:25 DEBUG client_cmd: /usr/bin/python3 -u pykms_Client.py 0.0.0.0 1688 -m Windows10 -n DummyClient -c ae3a27d1-b73a-4734-9878-70c949815218 -V DEBUG -F STDOUT
Fri, 17 Dec 2021 08:46:25 INFO Connecting to 0.0.0.0 on port 1688
Fri, 17 Dec 2021 08:46:25 INFO Connection successful !
Fri, 17 Dec 2021 08:46:25 INFO Connection accepted: 127.0.0.1:59122
Fri, 17 Dec 2021 08:46:25 INFO Sending RPC bind request...
Fri, 17 Dec 2021 08:46:25 INFO RPC bind request received.
Fri, 17 Dec 2021 08:46:25 DEBUG RPC Bind Request Bytes:

05000b13100000007400000001000000d016d0160000000002000000000001007521c8514e845047b0d8ec255555bc060100
0000045d888aeb1cc9119fe808002b10486002000000010001007521c8514e845047b0d8ec255555bc06010000002c1cb76c
12984045030000000000000001000000

Fri, 17 Dec 2021 08:46:25 DEBUG RPC Bind Request:

MSRPCHeader
ver_major: {5}
ver_minor: {0}
type: {11}
flags: {19}
representation: {16}
frag_len: {116}
auth_len: {0}
call_id: {1}
dataLen: {100}
pduData: {b'\xd0\x16\xd0\x16\x00\x00\x00\x00\x02\x00\x00\x00\x00\x00\x01\x00u!\xc8QN\x84PG\xb0\xd8\x
ec%UU\xbc\x06\x01\x00\x00\x00\x04]\x88\x8a\xeb\x1c\xc9\x11\x9f\xe8\x08\x00+\x10H`\x02\x00\x00\x00\x0
1\x00\x01\x00u!\xc8QN\x84PG\xb0\xd8\xec%UU\xbc\x06\x01\x00\x00\x00,\x1c\xb7l\x12\x98@E\x03\x00\x00\x
00\x00\x00\x00\x00\x01\x00\x00\x00'}
_pad: {0}
pad: {''}
_sec_trailer: {0}
sec_trailer: {''}
auth_dataLen: {0}
auth_data: {''}

MSRPCBind
max_tfrag: {5840}
max_rfrag: {5840}
assoc_group: {0}
ctx_num: {2}
Reserved: {0}
Reserved2: {0}
_ctx_items: {88}
ctx_items: {<pykms_RpcBind.MSRPCBind.CtxItemArray object at 0x149c8372a5e0>}

Fri, 17 Dec 2021 08:46:25 DEBUG RPC Bind Response:

MSRPCBindAck
ver_major: {5}
ver_minor: {0}
type: {12}
flags: {19}
representation: {16}
frag_len: {84}
auth_len: {0}
call_id: {1}
max_tfrag: {5840}
max_rfrag: {5840}
assoc_group: {274972479}
SecondaryAddrLen: {5}
SecondaryAddr: {'1688'}
Pad: {b'\x00'}
ctx_num: {2}
ctx_items: {b'\x00\x00\x00\x00\x04]\x88\x8a\xeb\x1c\xc9\x11\x9f\xe8\x08\x00+\x10H`\x02\x00\x00\x00\x
03\x00\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'}
sec_trailer: {b''}
auth_data: {b''}
pduData: {b''}
pad: {b''}

Fri, 17 Dec 2021 08:46:25 DEBUG RPC Bind Response Bytes:

05000c13100000005400000001000000d016d0163fbf631005003136383800000200000000000000045d888aeb1cc9119fe8
08002b10486002000000030003000000000000000000000000000000000000000000

Fri, 17 Dec 2021 08:46:25 INFO RPC bind acknowledged.
Fri, 17 Dec 2021 08:46:25 INFO RPC bind acknowledged.
Traceback (most recent call last):
File "/home/py-kms/pykms_Client.py", line 364, in <module>
clt_main(with_gui = False)
File "/home/py-kms/pykms_Client.py", line 280, in clt_main
client_create(clt_sock)
File "/home/py-kms/pykms_Client.py", line 221, in client_create
kmsRequest = createKmsRequest()
File "/home/py-kms/pykms_Client.py", line 318, in createKmsRequest
if clt_config['KMSProtocolMajorVersion'] == 4:
KeyError: 'KMSProtocolMajorVersion'
Fri, 17 Dec 2021 08:46:25 WARNING No data received.
Fri, 17 Dec 2021 08:46:25 INFO Connection closed: 127.0.0.1:59122
Fri, 17 Dec 2021 08:46:25 DEBUG sqlite_cmd: /usr/bin/python3 -u /home/sqlite_web/sqlite_web.py -H 0.0.0.0 --read-only -x /home/py-kms/db/pykms_database.db -p 8080
Fri, 17 Dec 2021 08:46:25 INFO Starting a dummy activation to ensure the database file is created
Fri, 17 Dec 2021 08:46:25 DEBUG client_cmd: /usr/bin/python3 -u pykms_Client.py 0.0.0.0 1688 -m Windows10 -n DummyClient -c ae3a27d1-b73a-4734-9878-70c949815218 -V DEBUG -F STDOUT
Fri, 17 Dec 2021 08:46:25 INFO Connecting to 0.0.0.0 on port 1688
Fri, 17 Dec 2021 08:46:25 INFO Connection successful !
Fri, 17 Dec 2021 08:46:25 INFO Connection accepted: 127.0.0.1:59122
Fri, 17 Dec 2021 08:46:25 INFO Sending RPC bind request...
Fri, 17 Dec 2021 08:46:25 INFO RPC bind request received.
Fri, 17 Dec 2021 08:46:25 DEBUG RPC Bind Request Bytes:

05000b13100000007400000001000000d016d0160000000002000000000001007521c8514e845047b0d8ec255555bc060100
0000045d888aeb1cc9119fe808002b10486002000000010001007521c8514e845047b0d8ec255555bc06010000002c1cb76c
12984045030000000000000001000000

Fri, 17 Dec 2021 08:46:25 DEBUG RPC Bind Request:

MSRPCHeader
ver_major: {5}
ver_minor: {0}
type: {11}
flags: {19}
representation: {16}
frag_len: {116}
auth_len: {0}
call_id: {1}
dataLen: {100}
pduData: {b'\xd0\x16\xd0\x16\x00\x00\x00\x00\x02\x00\x00\x00\x00\x00\x01\x00u!\xc8QN\x84PG\xb0\xd8\x
ec%UU\xbc\x06\x01\x00\x00\x00\x04]\x88\x8a\xeb\x1c\xc9\x11\x9f\xe8\x08\x00+\x10H`\x02\x00\x00\x00\x0
1\x00\x01\x00u!\xc8QN\x84PG\xb0\xd8\xec%UU\xbc\x06\x01\x00\x00\x00,\x1c\xb7l\x12\x98@E\x03\x00\x00\x
00\x00\x00\x00\x00\x01\x00\x00\x00'}
_pad: {0}
pad: {''}
_sec_trailer: {0}
sec_trailer: {''}
auth_dataLen: {0}
auth_data: {''}

MSRPCBind
max_tfrag: {5840}
max_rfrag: {5840}
assoc_group: {0}
ctx_num: {2}
Reserved: {0}
Reserved2: {0}
_ctx_items: {88}
ctx_items: {<pykms_RpcBind.MSRPCBind.CtxItemArray object at 0x149c8372a5e0>}

Fri, 17 Dec 2021 08:46:25 DEBUG RPC Bind Response:

MSRPCBindAck
ver_major: {5}
ver_minor: {0}
type: {12}
flags: {19}
representation: {16}
frag_len: {84}
auth_len: {0}
call_id: {1}
max_tfrag: {5840}
max_rfrag: {5840}
assoc_group: {274972479}
SecondaryAddrLen: {5}
SecondaryAddr: {'1688'}
Pad: {b'\x00'}
ctx_num: {2}
ctx_items: {b'\x00\x00\x00\x00\x04]\x88\x8a\xeb\x1c\xc9\x11\x9f\xe8\x08\x00+\x10H`\x02\x00\x00\x00\x
03\x00\x03\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'}
sec_trailer: {b''}
auth_data: {b''}
pduData: {b''}
pad: {b''}

Fri, 17 Dec 2021 08:46:25 DEBUG RPC Bind Response Bytes:

05000c13100000005400000001000000d016d0163fbf631005003136383800000200000000000000045d888aeb1cc9119fe8
08002b10486002000000030003000000000000000000000000000000000000000000

Fri, 17 Dec 2021 08:46:25 INFO RPC bind acknowledged.
Fri, 17 Dec 2021 08:46:25 INFO RPC bind acknowledged.
Traceback (most recent call last):
File "/home/py-kms/pykms_Client.py", line 364, in <module>
clt_main(with_gui = False)
File "/home/py-kms/pykms_Client.py", line 280, in clt_main
client_create(clt_sock)
File "/home/py-kms/pykms_Client.py", line 221, in client_create
kmsRequest = createKmsRequest()
File "/home/py-kms/pykms_Client.py", line 318, in createKmsRequest
if clt_config['KMSProtocolMajorVersion'] == 4:
KeyError: 'KMSProtocolMajorVersion'
Fri, 17 Dec 2021 08:46:25 WARNING No data received.
Fri, 17 Dec 2021 08:46:25 INFO Connection closed: 127.0.0.1:59122
Fri, 17 Dec 2021 08:46:25 DEBUG sqlite_cmd: /usr/bin/python3 -u /home/sqlite_web/sqlite_web.py -H 0.0.0.0 --read-only -x /home/py-kms/db/pykms_database.db -p 8080
Unable to open database file in read-only mode. Ensure that the database exists in order to use read-only mode.
```
Now the SQLITE WebUI is not working but the KMS Activation seems to work.

### LINKED ISSUE #49: Permission error setting up docker-compose via Docs
Just tried the current setup related into Docs with following docker-compose:

```yml
version: '3'

services:
  kms:
    image: ghcr.io/py-kms-organization/py-kms:python3
    ports:
      - 1688:1688
      - 8080:8080
    environment:
      - IP=0.0.0.0
      - SQLITE=true
      - HWID=RANDOM
      - LOGLEVEL=INFO
      - LOGFILE=/dev/stdout
    restart: always
    volumes:
      - ./db:/home/py-kms/db
      - /etc/localtime:/etc/localtime:ro
```
Resultet into this Output:
```
`Creating network "py-kms_default" with the default driver
Creating py-kms_kms_1 ... done
Attaching to py-kms_kms_1
kms_1  | Tue, 21 Dec 2021 04:59:32 INFO     Log level: INFO
kms_1  | Tue, 21 Dec 2021 04:59:32 INFO     Setting timzeone to America/Chicago
kms_1  | Tue, 21 Dec 2021 04:59:32 INFO     Setting gid to '1000'.
kms_1  | Tue, 21 Dec 2021 04:59:32 INFO     Setting uid to '100'.
kms_1  | Tue, 21 Dec 2021 04:59:32 INFO     Storing database file to /home/py-kms/db/pykms_database.db
kms_1  | Traceback (most recent call last):
kms_1  |   File "/home/py-kms/pykms_Server.py", line 640, in <module>
kms_1  |     server_main_no_terminal()
kms_1  |   File "/home/py-kms/pykms_Server.py", line 571, in server_main_no_terminal
kms_1  |     server_with_gui()
kms_1  |   File "/home/py-kms/pykms_Server.py", line 564, in server_with_gui
kms_1  |     root = pykms_GuiBase.KmsGui()
kms_1  |   File "/home/py-kms/pykms_GuiBase.py", line 68, in __init__
kms_1  |     tk.Tk.__init__(self, *args, **kwargs)
kms_1  |   File "/usr/lib/python3.9/tkinter/__init__.py", line 2270, in __init__
kms_1  |     self.tk = _tkinter.create(screenName, baseName, className, interactive, wantobjects, useTk, sync, use)
kms_1  | _tkinter.TclError: no display name and no $DISPLAY environment variable
kms_1  | 
kms_1  | During handling of the above exception, another exception occurred:
kms_1  | 
kms_1  | Traceback (most recent call last):
kms_1  |   File "/home/py-kms/pykms_Server.py", line 642, in <module>
kms_1  |     server_main_terminal()
kms_1  |   File "/home/py-kms/pykms_Server.py", line 543, in server_main_terminal
kms_1  |     server_check()
kms_1  |   File "/home/py-kms/pykms_Server.py", line 409, in server_check
kms_1  |     check_setup(srv_config, srv_options, loggersrv, where = "srv")
kms_1  |   File "/home/py-kms/pykms_Misc.py", line 549, in check_setup
kms_1  |     logger_create(logger, config, mode = 'a')
kms_1  |   File "/home/py-kms/pykms_Misc.py", line 179, in logger_create
kms_1  |     hand_rotate = RotatingFileHandler(filename = config['logfile'][0], mode = mode, maxBytes = int(config['logsize'] * 1024 * 512),
kms_1  |   File "/usr/lib/python3.9/logging/handlers.py", line 153, in __init__
kms_1  |     BaseRotatingHandler.__init__(self, filename, mode, encoding=encoding,
kms_1  |   File "/usr/lib/python3.9/logging/handlers.py", line 58, in __init__
kms_1  |     logging.FileHandler.__init__(self, filename, mode=mode,
kms_1  |   File "/usr/lib/python3.9/logging/__init__.py", line 1146, in __init__
kms_1  |     StreamHandler.__init__(self, self._open())
kms_1  |   File "/usr/lib/python3.9/logging/__init__.py", line 1175, in _open
kms_1  |     return open(self.baseFilename, self.mode, encoding=self.encoding,
kms_1  | PermissionError: [Errno 13] Permission denied: '/dev/stdout'
kms_1  | Tue, 21 Dec 2021 04:59:37 INFO     Starting a dummy activation to ensure the database file is created
kms_1  | Traceback (most recent call last):
kms_1  |   File "/home/py-kms/pykms_Client.py", line 364, in <module>
kms_1  |     clt_main(with_gui = False)
kms_1  |   File "/home/py-kms/pykms_Client.py", line 275, in clt_main
kms_1  |     client_check()
kms_1  |   File "/home/py-kms/pykms_Client.py", line 124, in client_check
kms_1  |     check_setup(clt_config, clt_options, loggerclt, where = "clt")
kms_1  |   File "/home/py-kms/pykms_Misc.py", line 549, in check_setup
kms_1  |     logger_create(logger, config, mode = 'a')
kms_1  |   File "/home/py-kms/pykms_Misc.py", line 179, in logger_create
kms_1  |     hand_rotate = RotatingFileHandler(filename = config['logfile'][0], mode = mode, maxBytes = int(config['logsize'] * 1024 * 512),
kms_1  |   File "/usr/lib/python3.9/logging/handlers.py", line 153, in __init__
kms_1  |     BaseRotatingHandler.__init__(self, filename, mode, encoding=encoding,
kms_1  |   File "/usr/lib/python3.9/logging/handlers.py", line 58, in __init__
kms_1  |     logging.FileHandler.__init__(self, filename, mode=mode,
kms_1  |   File "/usr/lib/python3.9/logging/__init__.py", line 1146, in __init__
kms_1  |     StreamHandler.__init__(self, self._open())
kms_1  |   File "/usr/lib/python3.9/logging/__init__.py", line 1175, in _open
kms_1  |     return open(self.baseFilename, self.mode, encoding=self.encoding,
kms_1  | PermissionError: [Errno 13] Permission denied: '/dev/stdout'`
```

Could "fix" the Permission issue on /dev/stdout temporarly by using a fixed location with .log ending within the docker-compose file to another folder but the DB issue stays.

### PR DESCRIPTION
**Pull request by simonmicro**
Fixes #48 
Fixes #49 (workaround)

### CODE REVIEW

File: docs/Getting Started.md [hunk 1]
```diff
@@ -52,7 +52,6 @@ services:
       - SQLITE=true
       - HWID=RANDOM
       - LOGLEVEL=INFO
-      - LOGFILE=/dev/stdout
```
Comment (Matthew-Beckett) [inline docs/Getting Started.md:55]: If we have issues with log to stdout we should provide a default value of a writable path.

File: py-kms/pykms_Client.py [hunk 2]
```diff
@@ -156,26 +156,34 @@ def client_check():
 def client_update():
         kmsdb = kmsDB2Dict()
 
+        loggerclt.debug(f'Searching in kms database for machine "{clt_config["mode"]}"...')
+
         appitems = kmsdb[2]
         for appitem in appitems:
                 kmsitems = appitem['KmsItems']
                 for kmsitem in kmsitems:                                
-                        name = re.sub('\(.*\)', '', kmsitem['DisplayName']).replace('2015', '').replace(' ', '')
```
Comment (Matthew-Beckett) [inline py-kms/pykms_Client.py:163]: Why the removals in this file?

**Review by Matthew-Beckett — CHANGES_REQUESTED** [review]
First review has some comments, mostly questions. Just one suggestion.

File: docs/Getting Started.md (hunk 1 above)
Comment (simonmicro) [reply docs/Getting Started.md:55]: I would not recommend that, as we already have historical problems with our current `/home/py-kms` path stuff...

Comment (Matthew-Beckett) [reply docs/Getting Started.md:55]: Is that because we run rootless?

File: py-kms/pykms_Client.py (hunk 2 above)
Comment (simonmicro) [reply py-kms/pykms_Client.py:163]: To fix the discovery for `Windows 10` configuration in our file, which contains only `Windows 10/11` information.

File: docs/Getting Started.md (hunk 1 above)
Comment (simonmicro) [reply docs/Getting Started.md:55]: Additionally the default value is already `STDOUT` - but of the process itself, see `start.py` - can we resolve this or do you still think we should set that?

Comment (simonmicro) [reply docs/Getting Started.md:55]: > Is that because we run rootless?

Yes. And because of the bug you have mentioned inside moby.

**Pull request merged**